File: common/src/main/java/com/pluxity/global/annotation/ResolvePermission.java
package com.pluxity.global.annotation;

import com.pluxity.user.entity.PermissionType;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
import org.springframework.stereotype.Component;

@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@Component
public @interface ResolvePermission {
    PermissionType value();
}

File: common/src/main/java/com/pluxity/global/annotation/ResolvePermission.kt
package com.pluxity.global.annotation

import com.pluxity.user.entity.PermissionType
import org.springframework.stereotype.Component

@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@Component
annotation class ResolvePermission(
    val value: PermissionType,
)
